### ISSUE TITLE
Add the rule `end-tag`

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#246]
Warn if there aren't end-tag no matter whether need it.

Incrrect:

```html
<!-- the li element is allowed to omit the end-tag -->
<ul><li>item</ul>

<!-- Warn so the li element is not allowed to omit the end-tag, of course -->
<body>
  <div>foo...
  <div>bar...
</body>
```
